pkgs/applications/misc/elogind/default.nix

```diff
@@ -0,0 +1,74 @@
+{ stdenv
+, lib
+, fetchFromGitHub
+, meson
+, ninja
+, m4
+, gperf
+, getent
+, libcap
+, gettext
+, pkgconfig
+, udev
+, eudev
+, libxslt
+, python3Packages
+, docbook5
+, docbook_xsl
+, docbook_xsl_ns
+, docbook_xml_dtd_42
+, docbook_xml_dtd_45
+
+# Defaulting to false because usually the rationale for using elogind is to
+# use it in situation where a systemd dependency does not work (especially
+# when building with musl, which elogind explicitly supports).
+, enableSystemd ? false
+}:
+
+with stdenv.lib;
+
+stdenv.mkDerivation rec {
+  pname = "elogind";
+  version = "239.5";
+
+  src = fetchFromGitHub {
+    owner = "elogind";
+    repo = pname;
+    rev = "v${version}";
+    sha256 = "1gdiy4vbx4gs2hnb79x14zi530mlq26glxpzp3c95w8l058wj4ba";
+  };
+
+  nativeBuildInputs = [
+    meson
+    ninja
+    m4
+    pkgconfig
+    gperf
+    getent
+    libcap
+    gettext
+    libxslt.bin # xsltproc
+    docbook5 docbook_xsl docbook_xsl_ns docbook_xml_dtd_42 docbook_xml_dtd_45 # needed for docbook without Internet
+    python3Packages.lxml # fixes: man/meson.build:111:0: ERROR: Could not execute command "/build/source/tools/xml_helper.py".
+  ];
+
+  buildInputs =
+    if enableSystemd then [udev] else [eudev];
+
+  preConfigure = ''
+    patchShebangs .
```

Comment:
ref. cross-compilation, you may patch a program that will end up on the host with the build time bash etc. Better to patch things individually.

Reply:
From IRC:

@nh2:

> for https://github.com/NixOS/nixpkgs/pull/72057#discussion_r339853874, I understand your argument against catch-all `patchShebangs .` for cross-compilation. But what should I use instead when I know it will run on the target machine?

@worldofpeace:

> you shouldn't have to do anything I think. but if you run patchShebangs once on the file/s and, for example, it gets build time bash, it won't run again on that file changing it to the other version. so what I do is run unpackPhase and patchShebangs in the unpacked archive and find what files just need to be patched for the build.

@nh2:

> but these files contain e.g. `/bin/sh`. Doesn't that mean they won't run on the target system?

@worldofpeace 

> I'm not sure I understand, that makes it sound like if you don't run patchShebangs at postPatch it isn't ran at all. and it's still going to happen in fixup, it's just interpreters that are already patched and in the nix store are going to be untouched when we reach that phase, and you'll have the wrong version of the interpreter

@nh2 

> ah, so you're saying that mkDerivation will patch all the _remaining_ files for me (so that they can run on the target) after the build phase is done? E.g. it will replace the shebangs of the remaining bash files by the bash version that run on the cross-compile target?

@worldofpeace 

> I believe so, the hook has fixupOutputHooks+=(patchShebangsAuto)

> with https://github.com/NixOS/nixpkgs/pull/43833 I believe it's actually correct now

> note it was reverted several times, but it's there now

> ok thanks, I will give a shot at minimising it then

Reply:
I have made the shebang patching more specific, and similar to how its done in the `systemd` derivation.